backend/core/rule/tree.py

```diff
@@ -9,32 +9,87 @@
 logger = logging.getLogger('backend')
 
 
+class Loader(yaml.Loader):
+    def __init__(self, stream):
+        self._root = os.path.split(stream.name)[0]
+        super(Loader, self).__init__(stream)
+
+    def include(self, node):
+        filename = os.path.join(self._root, self.construct_scalar(node))
+        with open(filename, 'r') as f:
+            return yaml.load(f, Loader)
+
+Loader.add_constructor('!include', Loader.include)
+
+
 class TreeLoader(object):
-    def __init__(self, rule_name, sugang_list, metadata):
-        f = open(os.getcwd() + '/rules/' + rule_name + '.yml', 'r', encoding='utf-8')
-        self.tree = yaml.load(f)
+    def __init__(self, rule_name, sugang_list, metadata, course_model):
+        base_dir = os.path.dirname(os.path.abspath(__file__))
+        f = open(os.path.join(base_dir, 'rules', rule_name + '.yml'), 'r', encoding='utf-8')
+        self.tree = yaml.load(f, Loader)
         f.close()
         self.sugang_list = sugang_list
         self.metadata = metadata
-        self.base_node = TreeNode(None, and_func(), '!GRADUATE', self.metadata)
+        self.course_model = course_model
+        self.base_node = TreeNode(None, and_func(), '!GRADUATE', self.metadata, course_model,
+                                  [0, 0, False], False, False)
         self.load_tree(self.tree, self.base_node)
 
     def load_tree(self, current_tree, previous_node):
         for course in current_tree:
             if isinstance(course, str):
-                previous_node.add_children(TreeNode(course, None, self.metadata, None, self.sugang_list))
+                if course.startswith('$'):
+                    subarea = course[1:]
+                    code_list = [item.code for item in self.course_model.objects.filter(subarea=subarea)]
+                    code_set = set(code_list)
+                    for code in code_set:
+                        previous_node.add_children(
+                            TreeNode(code, None, None, self.metadata, self.course_model,
+                                     [0, 0, False], False, False, self.sugang_list))
```

Comment:
I think this constructor parameters are messy.
I recommend use default parameters & keyword parameter passing.